packages/flutter/lib/src/widgets/media_query.dart

```diff
@@ -30,6 +30,22 @@ enum Orientation {
 /// If no [MediaQuery] is in scope then the [MediaQuery.of] method will throw an
 /// exception, unless the `nullOk` argument is set to true, in which case it
 /// returns null.
+///
+/// MediaQueryData includes two [EdgeInsets] values:
+/// [padding] and [viewInsets]. These
+/// values reflect the configuration of the device and are used by
+/// many top level widgets, like [SafeArea] and the cupertino and
+/// material scaffold widgets. The padding value defines areas that
+/// might not be completely visible, like the display "notch" on the
+/// iPhone X. The viewInsets value defines areas that aren't visible at
+/// all, typically because they're obscured by the device's keyboard.
+/// The viewInsets and padding values are independent, they're both
+/// measured from the edges of the MediaQuery widget's bounds. The
+/// bounds of the top level MediaQuery created by [WidgetsApp] are the
+/// same as the window that contains the app. The scaffolds typically
```

Comment:
Instead of making the scaffolds special, we should actually say all widgets that consumes/handles the paddings or insets should remove it in its descendant MediaQuery

Reply:
Done.

Reply:
> Might be worth saying too: if you make a custom widget and it consumes a padding/inset, you're responsible to make sure your child doesn't see it anymore. Otherwise, your child might decide to be funky.

This comment was meant to be here.